ark/utils/example_dataset_test.py

```diff
@@ -0,0 +1,273 @@
+
+    # Will cause duplicate downloads
+    def test_get_example_dataset(self, tmp_path_factory):
+        """
+        Tests to make sure that if an incorrect `dataset` is provided, the function
+        errors out with the appropriate error message for the user.
+        """
+
+        with pytest.raises(ValueError):
+            get_example_dataset("incorrect_dataset", save_dir=tmp_path_factory)
+
+    def test_check_empty_dst(self, tmp_path):
+        """
+        Tests to make sure that `ExampleDataset.check_empty_dst
+        ()` accurately
+        reports if a directory contains files or not.
+        """
+
+        example_dataset = ExampleDataset(None)
+        empty_data_dir: pathlib.Path = tmp_path / "empty_dst_dir"
+        packed_data_dir: pathlib.Path = tmp_path / "packed_dst_dir"
+        empty_data_dir.mkdir(parents=True)
+        packed_data_dir.mkdir(parents=True)
+
+        # Empty directory has no files
+        assert example_dataset.check_empty_dst(empty_data_dir) is True
+
+        # Directory has files
+        test_utils._make_blank_file(packed_data_dir, "data_test.txt")
+        assert example_dataset.check_empty_dst(packed_data_dir) is False
+
+    def _image_data_check(self, dir_p: pathlib.Path):
+        """
+        Checks to make sure that all the FOVs exist.
+
+        Args:
+            dir (pathlib.Path): The directory to check.
+        """
+        # Check to make sure all the FOVs exist
+        downloaded_fovs = list(dir_p.glob("*"))
+        downloaded_fov_names = [f.stem for f in downloaded_fovs]
+        assert set(self.fov_names) == set(downloaded_fov_names)
+
+        # Check to make sure all 22 channels exist
+        for fov in downloaded_fovs:
+            c_names = [c.stem for c in fov.rglob("*")]
+            assert set(self.channel_names) == set(c_names)
+
+    def _cell_table_check(self, dir_p: pathlib.Path):
+        """
+        Checks to make sure that the following cell tables exist:
+            * `cell_table_arcsinh_transformed.csv`
+            * `cell_table_size_normalized.csv`
+
+        Args:
+            dir_p (pathlib.Path): The directory to check.
+        """
+
+        downloaded_cell_tables = list(dir_p.glob("*.csv"))
+        downloaded_cell_table_names = [f.stem for f in downloaded_cell_tables]
+        assert set(self.cell_table_names) == set(downloaded_cell_table_names)
+
+    def _deepcell_output_check(self, dir_p: pathlib.Path):
+        """
+        Checks to make sure that all cell nucleus (feature 0) and cell membrane masks (feature 1)
+        exist from deepcell output.
+
+        Args:
+            dir_p (pathlib.Path): The directory to check.
+        """
+        downloaded_deepcell_output = list(dir_p.glob("*.tif"))
+        downloaded_deepcell_output_names = [f.stem for f in downloaded_deepcell_output]
+        assert set(self.deepcell_output_names) == set(downloaded_deepcell_output_names)
+
+    def _suffix_paths(self, dataset_download: ExampleDataset,
```

Comment:
The code throughout this file is organized really well, super understandable even though there's a lot here. Nice job!